src/ucp/core/ucp_context.c

```diff
@@ -1247,21 +1247,19 @@ static void ucp_free_resources(ucp_context_t *context)
 
 static ucs_status_t ucp_check_resource_config(const ucp_config_t *config)
 {
-     /* if we got here then num_resources > 0.
-      * if the user's device list is empty, there is no match */
      if ((0 == config->devices[UCT_DEVICE_TYPE_NET].count) &&
          (0 == config->devices[UCT_DEVICE_TYPE_SHM].count) &&
          (0 == config->devices[UCT_DEVICE_TYPE_ACC].count) &&
          (0 == config->devices[UCT_DEVICE_TYPE_SELF].count)) {
+         /* user's device list is empty, there is no match */
```

Comment:
i'd remove the comment altogether since the error message is already self-explanatory